src/appleseed-max-impl/appleseedblendmtl/appleseedblendmtl.cpp

```diff
@@ -0,0 +1,744 @@
+        MaterialCoat6               = 6,
+        MaterialCoat7               = 7,
+        MaterialCoat8               = 8,
+        MaterialCoat9               = 9,
+        MaterialCoat10              = 10,
+        MtlCount                 // keep last
+    };
+
+    enum TexmapId
+    {
+        // Changing these value WILL break compatibility.
+        TexmapMask1     = 0,
+        TexmapMask2     = 1,
+        TexmapMask3     = 2,
+        TexmapMask4     = 3,
+        TexmapMask5     = 4,
+        TexmapMask6     = 5,
+        TexmapMask7     = 6,
+        TexmapMask8     = 7,
+        TexmapMask9     = 8,
+        TexmapMask10    = 9,
+        TexmapCount                 // keep last
+    };
+
+    const MSTR g_texmap_slot_names[10] =
+    {
+        L"Mask 1",
+        L"Mask 2",
+        L"Mask 3",
+        L"Mask 4",
+        L"Mask 5",
+        L"Mask 6",
+        L"Mask 7",
+        L"Mask 8",
+        L"Mask 9",
+        L"Mask 10",
+    };
+
+    const MSTR g_material_slot_names[11] =
+    {
+        L"Base Material",
+        L"Layer Material 1",
+        L"Layer Material 2",
+        L"Layer Material 3",
+        L"Layer Material 4",
+        L"Layer Material 5",
+        L"Layer Material 6",
+        L"Layer Material 7",
+        L"Layer Material 8",
+        L"Layer Material 9",
+        L"Layer Material 10",
+    };
+
+    ParamBlockDesc2 g_block_desc(
+        // --- Required arguments ---
+        ParamBlockIdBlendMtl,                       // parameter block's ID
+        L"appleseedBlendMtlParams",                 // internal parameter block's name
+        0,                                          // ID of the localized name string
+        &g_appleseed_blendmtl_classdesc,            // class descriptor
+        P_AUTO_CONSTRUCT + P_AUTO_UI,               // block flags
+
+        // --- P_AUTO_CONSTRUCT arguments ---
+        ParamBlockRefBlendMtl,                      // parameter block's reference number
+
+        // --- P_AUTO_UI arguments for Blend rollup ---
+        IDD_FORMVIEW_PARAMS,                        // ID of the dialog template
+        IDS_FORMVIEW_PARAMS_TITLE,                  // ID of the dialog's title string
+        0,                                          // IParamMap2 creation/deletion flag mask
+        0,                                          // rollup creation flag
+        nullptr,                                    // user dialog procedure
+        
+        // --- Parameters specifications for Blend rollup ---
+
+        ParamIdBaseMtl, L"base_material", TYPE_MTL, P_SUBANIM | P_SHORT_LABELS, IDS_MATERIAL,
+            p_submtlno, 0,
+            p_ui, TYPE_MTLBUTTON,       IDC_MTLBTN_BASE,
+        p_end,
+
+        ParamIdLayerMtl, L"layer_material_list", TYPE_MTL_TAB, TexmapCount, P_SUBANIM | P_SHORT_LABELS, IDS_MATERIAL,
+            p_submtlno, 1,
+            p_ui, TYPE_MTLBUTTON,       IDC_MTLBTN_COAT_1, IDC_MTLBTN_COAT_2, IDC_MTLBTN_COAT_3,
```

Comment:
Didn't know you could pass lists here!

Reply:
Yes, there's also a way to change the size in runtime, but I didn't go that far.